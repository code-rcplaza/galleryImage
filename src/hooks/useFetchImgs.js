import { useEffect, useState } from "react"
import { getImgs } from '../helpers/getImgs'

export const useFetchImgs = (category) => {

    const [state, setstate] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        setTimeout(() => {
            getImgs(category)
                .then(gifs => {
                    setstate({
                        data: gifs,
                        loading: false
                    })
                })

        }, 1000);

    }, [category])

    return state
}
