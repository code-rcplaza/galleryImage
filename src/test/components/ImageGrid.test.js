import React from 'react';
import { shallow } from 'enzyme'
import ImagesGrid from '../../components/ImageGrid'
import { useFetchImgs } from '../../hooks/useFetchImgs';

jest.mock('../../hooks/useFetchImgs')

describe('Tests on <ImagesGrid />', () => {
    const category = 'photos'

    test('Should be correctly render ', () => {
        
        useFetchImgs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<ImagesGrid category={ category } />)

        expect( wrapper ).toMatchSnapshot()
    });

    test('Should show items when gifs are loaded with useFecthGif', () => {
        const imgs = [{
            albumId: 1,
            id: 1,
            title: "ASDF",
            url: "https://localhost/assets/image.jpg",
            thumbnailUrl: "https://localhost/assets/image.jpg",
        }]

        useFetchImgs.mockReturnValue({
            data: imgs,
            loading: false
        })

        const wrapper = shallow(<ImagesGrid category={ category } />)

        expect( wrapper.find('p').exists()).toBe(false)
        expect( wrapper.find('ImageGridItem').length).toBe(imgs.length)
    });
})