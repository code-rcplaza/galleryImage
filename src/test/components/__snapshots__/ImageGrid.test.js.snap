// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <ImagesGrid /> Should be correctly render  1`] = `
<Fragment>
  <Loading />
</Fragment>
`;
